**Question 3**

The one-step transition probability matrix P of a Markov chain with state space {0, 1} is given by:

P = [[1/4, 3/4], [1/2, 1/2]]

Assume that $P[X_0 = 0]$ = 1/3. Calculate $E[X_2]$

This is fairly straight forward. In the earlier questions, I have shown the evolution of transition matrix over many iterations as means to calculate probabilities of state after n-steps. Here I simply use that information. 

In [1]:
from __future__ import (absolute_import, division)
import numpy as np
from numpy.linalg import matrix_power

In [2]:
p = np.array([[1/4, 3/4], [1/2, 1/2]])
p_sqr = matrix_power(p, 2)
print(p_sqr)

[[ 0.4375  0.5625]
 [ 0.375   0.625 ]]


I also performed the task above by hand and obtained the same result. Now that I have the probabilities for the next point in discrete time,I can easily calculate the expectation values:

$E[X_2] = P[X_2=0] . 0 + P[X_2=1] . 1$

$ E[X_2] = P[X_0=0] . 0.5625 + P[X_0=1] . 0.625$
$ = 0.6042 $
